services/apps/alcs/src/alcs/board/board.service.ts

```diff
@@ -87,4 +93,61 @@ export class BoardService {
       },
     });
   }
+
+  async delete(code: string) {
+    const board = await this.getOneOrFail({
+      code,
+    });
+
+    //Un-link all statuses first
+    await this.boardStatusRepository.delete({
+      board: {
+        uuid: board.uuid,
+      },
+    });
+
+    await this.boardRepository.remove(board);
+  }
+
+  async create(createDto: BoardDto) {
+    await this.saveUpdates(new Board(), createDto);
+  }
+
+  async update(code: string, updateDto: BoardDto) {
+    const board = await this.getOneOrFail({
+      code,
+    });
+    await this.saveUpdates(board, updateDto);
+  }
+
+  private async saveUpdates(board: Board, updateDto: BoardDto) {
+    const cardTypes = await this.cardService.getCardTypes();
+
+    board.title = updateDto.title;
+    board.allowedCardTypes = updateDto.allowedCardTypes.map(
+      (code) => cardTypes.find((cardType) => cardType.code === code)!,
+    );
+    board.createCardTypes = updateDto.createCardTypes.map(
+      (code) => cardTypes.find((cardType) => cardType.code === code)!,
+    );
+
+    await this.boardRepository.save(board);
+
+    await this.boardStatusRepository.delete({
```

Comment:
a nitpick, it would be better to move this block into separate function, something like "setBoardStatuses". It will make it easier to read/maintain the code